rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/RxAudioPlayer.java

```diff
@@ -86,54 +89,86 @@ public int progress() {
      * play audio from local file. should be scheduled in IO thread.
      */
     public Single<Boolean> play(@NonNull final PlayConfig config) {
-        if (config.mType == PlayConfig.TYPE_FILE && config.mAudioFile != null &&
-                config.mAudioFile.exists()) {
-            return Single.create(new Single.OnSubscribe<Boolean>() {
-                @Override
-                public void call(final SingleSubscriber<? super Boolean> singleSubscriber) {
-                    stopPlay();
+        switch (config.mType) {
```

Comment:
I didn't clone the  repository, just downloaded the zip file.
So I have to commit on the web file by file. It's great if you can do this update.

In my opinion, the parameter check in  `PlayConfig` will make the code clean, but there wolud be two `switch(mType)` , one in `PlayConfig` , and one in `play()`.
Maybe `isArgumentIllegal()` can be implemented as 3 methods.